### ISSUE TITLE
Is there a rsample() (reparameterization) method like in pytorch's distribution module?

### ISSUE THREAD

**Issue by turmeric-blend** [tensorflow/probability#1077]
the reparameterization concept came from [Soft Actor Critic](https://arxiv.org/pdf/1801.01290.pdf).

[Exasmple of rsample()](https://pytorch.org/docs/stable/_modules/torch/distributions/normal.html#Normal.rsample)

**Comment by brianwa84** [comment]
Many distributions are reparameterized, just use the sample method.

FWIW, I think this concept comes from way earlier than SAC, eg at least
back to *Auto-Encoding Variational Bayes*, (Kingma & Welling, 2013)
<http://gregorygundersen.com/blog/2018/04/29/reparameterization/#kingma2013auto>,
and in some cases like gamma from more recent work Implicit
Reparameterization Gradients, Figurnov et al).

To find out if a distribution is reparameterized, you can look at
dist.reparameterization_type .

**Comment by turmeric-blend** [comment]
thanks for the clarification :100:
